test/data/test_multi_embedding_tensor.py

```diff
@@ -0,0 +1,75 @@
+import pytest
+import torch
+
+from torch_frame.data.multi_embedding_tensor import MultiEmbeddingTensor
+
+
+def test_size():
+    num_rows = 8
+    num_cols = 3
+    lengths = torch.tensor([1, 2, 3])
+    offset = torch.tensor([0, 1, 2, 3])
+    values = torch.rand((num_rows, lengths.sum().item()))
+    met = MultiEmbeddingTensor(
+        num_rows=num_rows,
+        num_cols=num_cols,
+        values=values,
+        offset=offset,
+    )
+
+    assert met.size(0) == num_rows
+    assert met.size(1) == num_cols
+    with pytest.raises(ValueError, match="not have a fixed length"):
+        met.size(2)
+    with pytest.raises(IndexError, match="Dimension out of range"):
+        met.size(3)
+
+    assert met.shape[0] == num_rows
+    assert met.shape[1] == num_cols
+    assert met.shape[2] == -1
+
+
+def test_cat():
+    num_rows = 3
+    num_cols = 2
+    offset = torch.tensor([0, 2, 4])
+    values = torch.tensor([
+        [0, 1, 2, 3, 4],
+        [5, 6, 7, 8, 9],
+        [10, 11, 12, 13, 14],
+    ], )
+    met = MultiEmbeddingTensor(
+        num_rows=num_rows,
+        num_cols=num_cols,
+        values=values,
+        offset=offset,
+    )
+
+    # Test row concat
+    met_cat = MultiEmbeddingTensor.cat([met, met], dim=0)
+    assert met_cat.num_rows == num_rows * 2
+    assert met_cat.num_cols == num_cols
+    expected_values = torch.tensor([
+        [0, 1, 2, 3, 4],
+        [5, 6, 7, 8, 9],
+        [10, 11, 12, 13, 14],
+        [0, 1, 2, 3, 4],
+        [5, 6, 7, 8, 9],
+        [10, 11, 12, 13, 14],
+    ], )
+    assert torch.allclose(met_cat.values, expected_values)
+    expected_offset = offset
+    assert torch.allclose(met_cat.offset, expected_offset)
+
+    # Test column concat
+    met_cat = MultiEmbeddingTensor.cat([met, met], dim=1)
+    assert met_cat.num_rows == num_rows
+    assert met_cat.num_cols == num_cols * 2
+    expected_values = torch.tensor([
+        [0, 1, 2, 3, 4, 0, 1, 2, 3, 4],
+        [5, 6, 7, 8, 9, 5, 6, 7, 8, 9],
+        [10, 11, 12, 13, 14, 10, 11, 12, 13, 14],
+    ], )
+    assert torch.allclose(met_cat.values, expected_values)
+    expected_offset = torch.tensor([0, 2, 4, 2, 4])
+    assert torch.allclose(met_cat.offset, expected_offset)
```

Comment:
Add more test cases to edge case, e.g. the tensors you'd like to concat are different shapes, different types, different device.

Reply:
Resolving this comment as I decided to add `cat` in a follow-up PR.